src/gam.py

```diff
@@ -5361,38 +5364,35 @@ def sendOrDropEmail(users, method='send'):
     elif myarg == 'subject':
       subject = sys.argv[i+1]
       i += 2
-    elif myarg == 'recipient':
+    elif myarg in ['recipient', 'to']:
       recipient = sys.argv[i+1]
       i += 2
-    elif myarg == 'sender':
+    elif myarg == 'from':
       sender = sys.argv[i+1]
       i += 2
-    elif myarg == 'labels':
-      labels = sys.argv[i+1].split(',')
+    elif myarg == 'header':
+      msgHeaders[sys.argv[i+1]] = sys.argv[i+2]
+      i += 3
+    elif method in ['insert', 'import'] and myarg == 'labels':
+      labels = shlexSplitList(sys.argv[i+1])
       i += 2
-    elif myarg == 'deleted':
-      if method not in ['insert', 'import']:
-        systemErrorExit(3, 'deleted is only valid on insertemail and importemail')
+    elif method in ['insert', 'import'] and myarg == 'deleted':
       kwargs['deleted'] = True
       i += 1
-    elif myarg == 'receivednow':
-      if method not in ['insert', 'import']:
-        systemErrorExit(3, 'received_now is only valid on insertemail and importemail')
-      kwargs['internalDateSource'] = 'receivedTime'
-      i += 1
-    elif myarg == 'checkspam':
-      if method not in ['import']:
-        systemErrorExit(3, 'check_spam is only valid on importemail')
-      kwargs['neverMarkSpam'] = False
+    elif myarg == 'date':
+      msgHeaders['Date'] = getTimeOrDeltaFromNow(sys.argv[i+1])
+      if method in ['insert', 'import']:
+        kwargs['internalDateSource'] = 'dateHeader'
+      i += 2
+    elif method == 'import' and myarg == 'nevercheckspam':
```

Comment:
API defaults to enabling check spam if this isn't set. That doesn't make a lot of sense to me, especially in the context of GAM here, if the admin is running the command to put the message in the user's mailbox, it almost certainly shouldn't go to Spam. I think I had this wrong but lets make neverMarkSpam=True the default for imports unless checkspam argument is supplied by user on GAM command line.